app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -154,51 +200,77 @@ const PermalinkPreview = ({embedData, showPermalinkPreviews, author, locale, tea
-                    size={32}
-                    showStatus={false}
-                />
-                <View style={styles.authorInfo}>
-                    <Text
-                        style={styles.authorName}
-                        numberOfLines={1}
-                    >
-                        {authorDisplayName}
-                    </Text>
-                    <FormattedTime
-                        timezone=''
-                        isMilitaryTime={false}
-                        value={post.create_at}
-                        style={styles.timestamp}
-                    />
+            <View style={styles.contentContainer}>
+                <View onLayout={handleContentLayout}>
+                    <View style={styles.header}>
+                        <ProfilePicture
+                            author={author || {id: post.user_id} as UserModel}
+                            size={32}
+                            showStatus={false}
+                        />
+                        <View style={styles.authorInfo}>
+                            <Text
+                                style={styles.authorName}
+                                numberOfLines={1}
+                            >
+                                {authorDisplayName}
+                            </Text>
+                            <FormattedTime
+                                timezone=''
+                                isMilitaryTime={false}
+                                value={post.create_at}
+                                style={styles.timestamp}
+                            />
+                        </View>
+                    </View>
+
+                    <View style={styles.messageContainer}>
+                        <Markdown
+                            baseTextStyle={styles.messageText}
+                            blockStyles={blockStyles}
+                            channelId={embedData.channel_id}
+                            disableGallery={true}
+                            disableHashtags={true}
+                            disableAtMentions={true}
+                            disableChannelLink={true}
+                            location={location}
+                            theme={theme}
+                            textStyles={textStyles}
+                            value={truncatedMessage}
+                        />
+                        {isEdited ? (
+                            <EditedIndicator
+                                baseTextStyle={styles.messageText}
+                                theme={theme}
+                                context={['paragraph']}
+                                iconSize={12}
+                                testID='permalink_preview.edited_indicator_separate'
+                            />
+                        ) : null}
+                    </View>
+
+                    {hasFiles && (
+                        <PermalinkFiles
```

Comment:
now I kind of struggle to see why this wrapper of `PermalinkFiles` is needed.

Reply:
The `PermalinkFiles` wrapper is needed for viewport event bridging. The Files component listens for `ITEM_IN_VIEWPORT` events to trigger image loading, but in permalink previews, there's a parent-child relationship:
- Parent: Permalink preview container (`${parentLocation}-${parentPostId}`)
- Child: Files component (`${location}-${post.id}`)

The wrapper listens for parent viewport events and propagates them to the child Files component, ensuring images load properly when the permalink preview comes into view. Without this bridge, the Files component would never receive viewport signals, and images wouldn't load.

For the Post component, it doesn't need this wrapper because it's the top-level component that receives viewport events directly.